src/EPPlus/Table/PivotTable/Filter/ExcelPivotTableFilter.cs

```diff
@@ -378,6 +448,203 @@ internal void CreateValueFilter()
+    {
+        var value = pivotTable.Fields[Fld].Cache.SharedItems[index].ToString();
+        switch (Type)
+        {
+            //Caption filters (String)
+            case ePivotTableFilterType.CaptionEqual:
+                return value.Equals(StringValue1, StringComparison.InvariantCultureIgnoreCase);
+            case ePivotTableFilterType.CaptionNotEqual:
+                return !value.Equals(StringValue1, StringComparison.InvariantCultureIgnoreCase);
+            case ePivotTableFilterType.CaptionBeginsWith:
+                return value.StartsWith(StringValue1, StringComparison.InvariantCultureIgnoreCase);
+            case ePivotTableFilterType.CaptionNotBeginsWith:
+                return !value.StartsWith(StringValue1, StringComparison.InvariantCultureIgnoreCase);
+            case ePivotTableFilterType.CaptionEndsWith:
+                return value.EndsWith(StringValue1, StringComparison.InvariantCultureIgnoreCase);
+            case ePivotTableFilterType.CaptionNotEndsWith:
+                return !value.EndsWith(StringValue1, StringComparison.InvariantCultureIgnoreCase);
+            case ePivotTableFilterType.CaptionGreaterThan:
+                return string.Compare(value, StringValue1, true) > 0;
+            case ePivotTableFilterType.CaptionGreaterThanOrEqual:
+                return string.Compare(value, StringValue1, true) >= 0;
+            case ePivotTableFilterType.CaptionLessThan:
+                return string.Compare(value, StringValue1, true) < 0;
+            case ePivotTableFilterType.CaptionLessThanOrEqual:
+                return string.Compare(value, StringValue1, true) <= 0;
+            case ePivotTableFilterType.CaptionContains:
+                return value.IndexOf(StringValue1, StringComparison.InvariantCultureIgnoreCase) >= 0;
+            case ePivotTableFilterType.CaptionNotContains:
+                return value.IndexOf(StringValue1, StringComparison.InvariantCultureIgnoreCase) < 0;
+            case ePivotTableFilterType.CaptionBetween:
+                return string.Compare(value, StringValue1, true) >= 0 && string.Compare(value, StringValue2, true) <= 0;
+            case ePivotTableFilterType.CaptionNotBetween:
+                return !(string.Compare(value, StringValue1, true) >= 0 && string.Compare(value, StringValue2, true) <= 0);
+            case ePivotTableFilterType.Unknown:
+            default:
+                return false;
+        }
+    }
+        /// <summary>
+        /// Handle date filters. This is filter enum values below 100.
+        /// </summary>
+        /// <param name="pivotTable"></param>
+        /// <param name="index"></param>
+        /// <returns></returns>
+        private bool MatchDate(ExcelPivotTable pivotTable, int index)
+        {
+            var date = ConvertUtil.GetValueDate(pivotTable.Fields[Fld].Cache.SharedItems[index]);
+            var value1Date = ConvertUtil.GetValueDate(Value1);
+            if (date.HasValue && value1Date.HasValue)
```

Comment:
Switch case should have a default for "hängslen och livrem"

Reply:
Added default: with Exception